torch/fx/traceback.py

```diff
@@ -291,6 +293,43 @@ def annotate(annotation_dict: dict):
             del current_meta["custom"]
 
 
+@compatibility(is_backward_compatible=False)
+def annotate_fn(annotation_dict: dict):
```

Comment:
Are the dict keys always strings? If so, that should be specified. Also we need to specify it returns a decorator, or a Callable that returns a Callable preserving the args, kwargs, and return type.